app/views/layouts/application.html.erb

```diff
@@ -19,9 +19,12 @@
   <!-- header -->
   <header>
     <%= render 'shared/nav' %>
-    <div class="container">
-      <%= render partial: '/shared/flash_msg', layout: nil %>
-    </div>
+    <!-- if an organization header is displayed, flash messages will instead display in _organization_header.html.erb -->
+    <% if !defined? @organization %>
```

Comment:
Checking for the existence of the @organization instance variable may not be a reliable way to check whether the organization header is rendered. I'd suggest refactoring how the organization header is rendered to address this issue. One way to do this would be to render the organization header within the header tag in `layouts/application.html.erb`. Something like:

```
  <header>
    <%= render 'shared/nav' %>
    <%= yield :org_header %>
      <div class="container">
        <%= render partial: '/shared/flash_msg', layout: nil %>
      </div>
```

You would then wrap the the `organization_header.html.erb` partial in a `content_for` block so it will render when the named yield is called:

```
<% content_for :org_header do %>
  <div class="position-relative overflow-hidden mb-5 pb-2 pt-4 bg-light organization border-bottom">
  #etc.
<% end %>
```

With this approach the org header is always rendered before the flash message so you don't have to worry about trying to figure out if the org header is present or not to conditionally display the flash message in different locations. Happy to pair on this if you have questions.

Reply:
Rails docs on `content_for`:

https://guides.rubyonrails.org/layouts_and_rendering.html#using-the-content-for-method

Reply:
@corylown Thank you for the detailed tips. I've refactored to use content_for and yield. I also removed the extra `<%= yield %> ` from the organization_header, as I think with the new pattern it doesn't actually do anything. Let me know if this is along the lines of your expectation.